content/3_Silver/2P.mdx

```diff
@@ -31,6 +31,143 @@ frequency: 2
 	</Resource>
 </Resources>
 
+## Two Pointers
+
+The two pointers method iterates two pointers across an array, to track the start and end
+of an interval. It can also be used to track two values in an array as shown in CPH's 2SUM solution. 
+
+<FocusProblem problem="book" />
+
+## Solution - Books
+
+We want to find the longest contiguous segment of books that can be read within $t$ minutes. 
+
+To accomplish this, we can define $\texttt{left}$ and $\texttt{right}$ as the start and end of the segment respectively.
+
+Since being able to read segment $\texttt{left} \dots \texttt{right}$ implies we can read segment $\texttt{left} \dots \texttt{right}-1$, we can use two-pointers (or, alternatively, [binary search](/silver/binary-search)).
+
+For every $\texttt{left}$ index, we need to increase our $\texttt{right}$ pointer until the total time for that segment exceeds $t$.
```

Comment:
> For every $\texttt{left}$ index, we need to increase our $\texttt{right}$ pointer until the total time for that segment exceeds $t$.
Kind of misleading - you're not really increasing unit it exceeds. It's more like increasing until it reaches max.

Reply:
wdym by the max? im increasing it until it exceeds t right?

Reply:
well it isn't really _exceeding_ t right - it's more like increasing until it hits the maximum that's less than t.

Reply:
> well it isn't really _exceeding_ t right - it's more like increasing until it hits the maximum that's less than t.

@andrewwangva oh i c what u mean, but for each time step, we are increasing right until it exceeds t tho.

Reply:
oh wait i misread ur msg, ill make it clearer

Reply:
okay fixed.